Description: fix type error
 .
Author: xiao sheng wen <atzlinux@sina.com>

---
Last-Update: 2020-04-19

diff --git a/src/ec_throw.cpp b/src/ec_throw.cpp
index fdf0551..f4511e6 100644
--- a/src/ec_throw.cpp
+++ b/src/ec_throw.cpp
@@ -24,7 +24,7 @@ std::runtime_error throw_except(const char* func, const char* file, long line, c
 {
 	std::stringstream stream;
 	stream << "File: " 	<< file << std::endl;
-	stream << "Fucntion: " 	<< func << std::endl;
+	stream << "Function: " 	<< func << std::endl;
 	stream << "Expression:"	<< std::endl << line << ":\t" << var << std::endl; 
 	stream	<< "Errno: "	<< errno << std::endl;
 	stream << strerror(errno) << std::endl; 
