Description: type error fix
Author: xiao sheng wen <atzlinux@sina.com>

---
Last-Update: 2020-04-20 
===================================================================
--- a/doc/convert_pgn.6.in
+++ b/doc/convert_pgn.6.in
@@ -20,7 +20,7 @@
 .fi
 .SH DESCRIPTION
 The \fBconvert_pgn\fP program converts the format of characters in \fIinput-file\fP, from
-one formated character set to another. The result is written to default file
+one formatted character set to another. The result is written to default file
 "/tmp/gmchess.pgn" unless otherwise specified by the \fIoutput-file\fP option.
 .RE
 .PP
