commit efbfc0338ce12e7ce9dce09d786ca8e1958daf4d
Author: Boyuan Yang <073plan@gmail.com>
Date:   Thu Jan 25 21:19:47 2018 +0800

    src: Engine.cc: Fix a fprintf format error with DLOG

diff --git a/src/Engine.cc b/src/Engine.cc
index d316c0e..dd0187e 100644
--- a/src/Engine.cc
+++ b/src/Engine.cc
@@ -2098,7 +2098,7 @@ uint32_t Engine::hanzi_to_iccs(uint32_t f_hanzi)
 							n++;
 						}
 					}
-					DLOG(" n=%d\n");
+					DLOG(" n=%d\n", n);
 					/** 为纵线上的棋子排序*/
 					for(int k=start;k<n-1;k++)
 						for(int j=n-2;j>=k;j--){
