commit b341b0da07f72d6b6c989de42c7c525ddbf1513f
Author: Boyuan Yang <073plan@gmail.com>
Date:   Wed Feb 7 18:20:12 2018 +0800

    src: Board.cc: prevent path buffer

diff --git a/src/Board.cc b/src/Board.cc
index 4317d17..642418f 100644
--- a/src/Board.cc
+++ b/src/Board.cc
@@ -175,17 +175,19 @@ void Board::set_themes(const std::string& themes_)
 }
 Glib::RefPtr<Gdk::Pixbuf> Board::get_pic(const std::string& name_)
 {
-	char path[1024] = {0};
-	sprintf(path, DATA_DIR"/themes/%s/%s",theme.c_str(),name_.c_str());
-	return Gdk::Pixbuf::create_from_file(path);
+    const std::string path = \
+            std::string(DATA_DIR) + \
+            "/themes/" + theme + "/" + name_;
+    return Gdk::Pixbuf::create_from_file(path);
 
 }
 
 Glib::RefPtr<Gdk::Pixbuf> Board::get_spic(const std::string& name_)
 {
-	char path[1024] = {0};
-	sprintf(path, DATA_DIR"/themes/%s-small/%s",theme.c_str(),name_.c_str());
-	return Gdk::Pixbuf::create_from_file(path);
+    const std::string path = \
+            std::string(DATA_DIR) + \
+            "/themes/" + theme + "-small/" + name_;
+    return Gdk::Pixbuf::create_from_file(path);
 
 }
 
